konnectivity-client/pkg/client/client.go

```diff
@@ -339,7 +339,20 @@ func (t *grpcTunnel) serve(tunnelCtx context.Context) {
 			conn, ok := t.conns.get(resp.ConnectID)
 
 			if !ok {
-				klog.V(1).InfoS("Connection not recognized", "connectionID", resp.ConnectID)
+				klog.ErrorS(nil, "Connection not recognized", "connectionID", resp.ConnectID, "packetType", "DATA")
+				req := &client.Packet{
```

Comment:
No point in trying to send a close request if the ConnectID == 0.

Reply:
The connection ID is only removed from `t.conns` when receiving a CLOSE_RSP from the server, not for a client-side close. So the only way this would be hit is if the server sent a DATA packet _after_ a CLOSE_RSP, which should never happen.

Note that I did change the CLOSE_RSP version of this log (line 374) to an info log, as there are cases where multiple CLOSE_RSP packets might be sent.

Reply:
I don't think we'd ever send a DATA packet with a `ConnectID == 0`? We could check for it here, but then we should check for it everywhere, but that seems unnecessary.

Reply:
Per discussion, added this as an extra safety measure.